service/src/java/org/apache/hive/service/cli/thrift/ThriftBinaryCLIService.java

```diff
@@ -100,7 +102,15 @@ protected void initServer() {
 
       // Server args
       int maxMessageSize = hiveConf.getIntVar(HiveConf.ConfVars.HIVE_SERVER2_THRIFT_MAX_MESSAGE_SIZE);
-      TThreadPoolServer.Args sargs = new TThreadPoolServer.Args(serverSocket).processorFactory(processorFactory)
+      TThreadPoolServer.Args sargs = new TThreadPoolServer.Args(new TServerSocket(serverSocket.getServerSocket()) {
```

Comment:
why don't we reuse `MetastoreTServerSocket`, maybe we could use some generic name?

Reply:
sorry, i did catch your point. 
If I read it correctly, the anonymous class looks similar to MetastoreTServerSocket except for the `keepAlive`.
Can't we extract it to the common module and reuse (pass keepAlive & maxMessageSize as constructor args)?